app/src/main/java/com/amaze/filemanager/filesystem/compressed/RarArchiveExt.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2014-2021 Arpit Khurana <arpitkh96@gmail.com>, Vishal Nehra <vishalmeham2@gmail.com>,
+ * Emmanuel Messulam<emmanuelbendavid@gmail.com>, Raymond Lai <airwave209gt at gmail.com> and Contributors.
+ *
+ * This file is part of Amaze File Manager.
+ *
+ * Amaze File Manager is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.amaze.filemanager.filesystem.compressed
+
+import android.os.Build.VERSION.SDK_INT
+import android.os.Build.VERSION_CODES.M
+import com.github.junrar.Archive
+
+/**
+ * Extension function to patch [Archive.isPasswordProtected] which uses API that is not available
+ * for Android 6.0 or lower.
+ *
+ * @see [Archive.isPasswordProtected]
+ * @see [java.util.stream.Stream]
+ * @return if archive is password protected
+ */
+fun Archive.isPasswordProtectedCompat(): Boolean {
```

Comment:
There's a generic extension kotlin file already. I don't think it's required to create another Rar specific kotlin file.

Reply:
Against, please keep the functions in the package they belong to, aka this in compressed.

Reply:
Ok even then a separate file for rar extension function is definitely not required. It's just too much segregation.